sn_client/src/wallet/mod.rs

```diff
@@ -6,30 +6,56 @@
 // KIND, either express or implied. Please review the Licences for the specific language governing
 // permissions and limitations relating to use of the SAFE Network Software.
 
-pub(crate) mod send_client;
-pub(crate) mod verifying_client;
-
 use super::Client;
 
+use sn_transfers::{
+    client_transfers::TransactionOutputs,
+    wallet::{Error, LocalWallet, Result},
+};
+
+use futures::future::join_all;
 use sn_dbc::{Dbc, PublicAddress, Token};
-use sn_transfers::wallet::{Error, LocalWallet, Result, SendWallet};
 
 /// A wallet client can be used to send and
 /// receive tokens to/from other wallets.
-pub struct WalletClient<W: SendWallet> {
+pub struct WalletClient {
     client: Client,
-    wallet: W,
+    wallet: LocalWallet,
+    /// These have not yet been successfully confirmed in
+    /// the network and need to be republished, to reach network validity.
+    /// We maintain the order they were added in, as to republish
+    /// them in the correct order, in case any later spend was
+    /// dependent on an earlier spend.
+    unconfirmed_txs: Vec<TransactionOutputs>,
```

Comment:
This logic was in the `LocalWallet`, moved it to client code since it requires Network activity aka Client.